### PR TITLE
Feature: set DSOJobDistributionInfo's 'AllowedDistributiontypes' with analyze_setup()

### PR DESCRIPTION
**Pull request by nikolarobottesla**
needed to modify update_hpc_option() to set this property

### CODE REVIEW

**Review by maxcapodi78 — APPROVED** [review]
LGTM overall. To autofix style errors I suggest the following.
Run in your venv:
- pip install -r requrirements_test.txt
- pre-commit install

File: pyaedt/application/Analysis.py [hunk 1]
```diff
@@ -1653,6 +1654,9 @@ def analyze_setup(
             Either if use or not auto settings in task/cores. It is not supported by all Setup.
         num_variations_to_distribute : int, optional
             Number of variations to distribute. For this to take effect ``use_auto_settings`` must be set to ``True``.
+        allowed_distribution_types : list, optional
+            List of strings, each string represents a distribution type, The default ``None`` does nothing. An empty
```
Comment (MaxJPRey) [inline pyaedt/application/Analysis.py:1658]:
```suggestion
            List of strings. Each string represents a distribution type. The default value ``None`` does nothing. An empty
```

File: pyaedt/application/Analysis.py [hunk 2]
```diff
@@ -1653,6 +1654,9 @@ def analyze_setup(
             Either if use or not auto settings in task/cores. It is not supported by all Setup.
         num_variations_to_distribute : int, optional
             Number of variations to distribute. For this to take effect ``use_auto_settings`` must be set to ``True``.
+        allowed_distribution_types : list, optional
+            List of strings. Each string represents a distribution type. The default value ``None`` does nothing. An empty
+            list ``[]`` disables all types.
```
Comment (maxcapodi78) [inline pyaedt/application/Analysis.py:1659]:
```suggestion
            List of strings. Each string represents a distribution type. The default value ``None`` does nothing.
            An empty list ``[]`` disables all types.
```

**Review by maxcapodi78 — APPROVED** [review]
LGTM

**Review by MaxJPRey — APPROVED** [review]
LGTM. Thanks for contributing @nikolarobottesla !

**Comment by nikolarobottesla** [comment]
> 

thanks, should I add 'pre-commit' package to the requirements_test.txt?

**Comment by MaxJPRey** [comment]
@nikolarobottesla No it is not necessary to add it to `requirements_test.py`.
As soon as you will have run `pre-commit` on your machine once, it should be running next time automatically.

**Comment by MaxJPRey** [comment]
@nikolarobottesla Feel free to "squash and merge" whenever you think it is all set.
Thanks again for your work.

**Review by maxcapodi78 — APPROVED** [review]

**Pull request merged**